### PR TITLE
Adopt lintrunner and enable github actions lint checks 1/2

### PR DESCRIPTION
**Pull request by justinchuby**
## Type of Change

CI

## Description

- Adopt lintrunner as standard ONNX tooling and enable github actions lint checks.
- Use black and isort to format the code base. The CI will enforce formatting from now on.
- Removed `.pre-commit-config.yaml`. Checks can be supported with lintrunner.

This is the first of two PRs. The second PR will enable more ruff rules and apply autofixes.

Fixes https://github.com/onnx/neural-compressor/issues/5

## Expected Behavior & Potential Risk

Auto-formatted code.

## How has this PR been tested?

CI

## Dependency Change?

+dev dependencies: lintrunner, lintrunner-adapters

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 [tool.isort]
 profile = "black"
 line_length = 120
-known_first_party = ["neural_compressor"]
```
Comment (justinchuby) [inline pyproject.toml:4]: This is automatic.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -46,12 +46,6 @@ repos:
               examples/.+
           )$
 
-  - repo: https://github.com/asottile/yesqa
-    rev: v1.5.0
-    hooks:
-      - id: yesqa
-        name: Unused noqa
```
Comment (justinchuby) [inline .pre-commit-config.yaml:53]: This is handled by ruff

**Comment by chensuyue** [comment]
@justinchuby, Although I got the admin permission, I am not able to install this APP.   
image

If the tool is not allowed to be installed, let's remove ".pre-commit-config.yaml" directly.

**Comment by justinchuby** [comment]
SGTM. Just did

**Pull request merged**